homeassistant/helpers/service.py

```diff
@@ -630,12 +636,34 @@ async def async_get_all_descriptions(
 
                 # Don't warn for missing services, because it triggers false
                 # positives for things like scripts, that register as a service
+                #
+                # When name & description are in the translations use those;
+                # otherwise fallback to backwards compatible behavior from
+                # the time when we didn't have translations for descriptions yet.
+                # This mimics the behavior of the frontend.
                 description = {
-                    "name": yaml_description.get("name", ""),
-                    "description": yaml_description.get("description", ""),
+                    "name": translations.get(
+                        f"component.{domain}.services.{service_name}.name",
+                        yaml_description.get("name", ""),
```

Comment:
Should we fallback to service name if there is no name?

```suggestion
                        yaml_description.get("name", service_name),
```

Reply:
Hm nevermind, we did empty string before too